indico/modules/events/registration/controllers/management/fields.py

```diff
@@ -130,15 +132,11 @@ class RHRegistrationFormAddField(RHManageRegFormSectionBase):
 
     def _process(self):
         field_data = snakify_keys(request.json['fieldData'])
-        if field_data['input_type'] == 'date':
-            date_format = field_data['date_format'].split(' ')
-            field_data['date_format'] = date_format[0]
-            if len(date_format) == 2:
-                field_data['time_format'] = date_format[1]
-        elif field_data['input_type'] == 'radio':
-            items = field_data['radioitems']
-            for item in items:
-                item['id'] = unicode(uuid4())
+        try:
+            field_impl_type = get_field_types()[field_data['input_type']]
+            field_impl_type.modify_post_data(field_data)
+        except KeyError:
```

Comment:
You also need to run the label-specific code before getting here - labels have no field implementation